lib/internal/assert/assertion_error.js

```diff
@@ -437,6 +437,23 @@ class AssertionError extends Error {
     this.name = 'AssertionError';
   }
 
+  get details() {
+    return this;
+  }
+
+  set details(details) {
+    delete this.actual;
```

Comment:
Thinking about this further, I don't think this getter and setter is needed or ideal. Setting the details property equal to the array returned by report is enough.